tests/backup/backup_kdmp_test.go

```diff
@@ -0,0 +1,770 @@
+							log.Infof(fmt.Sprintf("the list of files created in mountPath [%v] for pod [%v]: %v", mountPath, pod.Name, fileList))
+							log.Infof(fmt.Sprintf("the list of directories created in mountPath [%v] for pod [%v]: %v", mountPath, pod.Name, dirList))
+							fileListMountMap[mountPath] = fileList
+							dirListMountMap[mountPath] = dirList
+							finalFileList[mountPath] = append(finalFileList[mountPath], fileList...)
+							finalFileList[mountPath] = append(finalFileList[mountPath], dirList...)
+						}
+					}
+				}
+			}
+		})
+
+		Step("Taking manual backup of namespaces with rules without excluding any files or directories", func() {
+			log.InfoD(fmt.Sprintf("Taking manual backup of namespaces with rules without excluding any files or directories"))
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			backupNames = make([]string, 0)
+			for _, namespace := range bkpNamespaces {
+				backupName = fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+				err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, orgID, clusterUid, preRuleName, preRuleUid, postRuleName, postRuleUid)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+				backupNames = append(backupNames, backupName)
+				backupNamespaceMap[backupName] = namespace
+			}
+		})
+
+		Step("Taking restore of backups without excluding any files or directories", func() {
+			log.InfoD("Taking restore of backups without excluding any files or directories")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoredNamespaces = make([]string, 0)
+			for _, backupName := range backupNames {
+				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{backupNamespaceMap[backupName]})
+				restoreName = fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+				restoreNamespace := "custom3-" + backupNamespaceMap[backupName]
+				namespaceMapping := map[string]string{backupNamespaceMap[backupName]: restoreNamespace}
+				restoredNamespaces = append(restoredNamespaces, restoreNamespace)
+				err = CreateRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), destinationClusterName, orgID, appContextsToBackup)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+
+		Step("List files and directories from the mount path after restore and verify all files and directories are present", func() {
+			log.InfoD("List files and directories from the mount path after restore and verify all files and directories are present")
+
+			defer func() {
```

Comment:
Should we move the defer to JustAfterEach?

Reply:
its mandatory to for each steps to have err := SetSourceKubeConfig() , adding to JustAfterEach will be an additional step.

Reply:
@kshithijiyer-px , verified multiple defers are not called in case of any failure. 
please find the log.
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Adilk/job/kdmp-byoc/55/console